src/Vision/VNRecognizedPointsObservation.cs

```diff
@@ -0,0 +1,71 @@
+//
+// VNRecognizedPointsObservation.cs
+//
+// Authors:
+//	Alex Soto  <alexsoto@microsoft.com>
+//
+// Copyright (c) Microsoft Corporation.
+//
+
+using System;
+using Foundation;
+using ObjCRuntime;
+
+namespace Vision {
+	public partial class VNRecognizedPointsObservation {
+
+		public T [] GetAvailableKeys<T> () where T : Enum
+		{
+			var type = typeof (T);
+			if (!(type == typeof (VNHumanBodyPoseObservationJointName) || type == typeof (VNHumanHandPoseObservationJointName)))
+				throw new InvalidOperationException ($"Only '{nameof (VNHumanBodyPoseObservationJointName)}' and '{nameof (VNHumanHandPoseObservationJointName)}' are supported.");
+
+			NSString[] keys = AvailableKeys;
+			if (keys == null)
+				return null;
+
+			var count = keys.Length;
+			var availableKeys = new T [count];
+
+			if (count == 0)
+				return availableKeys;
+
+			if (availableKeys is VNHumanBodyPoseObservationJointName) {
```

Comment:
Idea to make this simpler:

```csharp
if (T is VNHumanBodyPoseObservationJointName)
    availableKeys = Array.Convert (keys, (v) => (T) (object) VNHumanBodyPoseObservationJointNameExtensions.GetValue (v));
else if (T is VNHumanHandPoseObservationJointName)
    availableKeys = Array.Convert (keys, (v) => (T) (object) VNHumanHandPoseObservationJointNameExtensions.GetValue (v));
```

Reply:
AFAIK you can even use a switch statement:
```csharp
switch (T)
{
        case VNHumanBodyPoseObservationJointName:
            availableKeys = Array.Convert (keys, (v) => (T) (object) VNHumanBodyPoseObservationJointNameExtensions.GetValue (v));
        case VNHumanHandPoseObservationJointName:
            availableKeys = Array.Convert (keys, (v) => (T) (object) VNHumanHandPoseObservationJointNameExtensions.GetValue (v));
}
```

Reply:
You can't really pattern match with 'T'